hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/WriteStatus.java

```diff
@@ -181,6 +188,15 @@ private void updateStatsForFailure() {
     totalErrorRecords++;
   }
 
+  public void removeMetadataStats() {
+    this.writtenRecordDelegates.clear();
+    this.stat.removeRecordStats();
```

Comment:
Not sure if we can unify the index stats in one class `IndexStats`: record delegates/record stats.

Reply:
currently we are reusing existing data structures.
since in first phase, we only plan to target RLI for streaming writes. 

in subsequent patches, we will be adding secondary index, col stats etc. 
So, when we plan to add more stats, we will enhance this to move all stats to something called IndexStats. 
currently, it does not makes sense to just move one entry i.e. writtenRecordDelegates to a new class.

Reply:
I think it would be better if we take it up in a separate jira.